[native] Add dockerfiles for dependencies and runtime and remove existing dockerfiles (#21929)
prestodb/presto: presto-native-execution/scripts/dockerfiles/README.md

# Prestissimo - Dockerfile build

> 📝 _**Note:** Please post in Presto Slack if you have any questions_

> There are two kinds of dockerfiles: 

Reviewer: ```suggestion
There are two kinds of Dockerfiles: 
```
Standard seems to be "Dockerfile" as one word and with initial capitalization. I'll mark these throughout, but please let me know if you have a different understanding of how to commonly refer to this item.

1) A platform specific file to build dependencies. Current supported platforms are
   1) Centos-8-stream with gcc9.
   2) Ubuntu-22.04 with gcc11.
2) A platform agnostic file to build Prestissimo runtime on top of the dependency image.
### Dependency Dockerfiles

> These dockerfile installs all the dependencies including those needed for testing.

Reviewer: ```suggestion
These Dockerfiles install all the dependencies including those needed for testing.
```

A list of dependencies can be found [here](../../README.md).
By default, the dependencies are built in Release mode.

> This image needs to be built only when some dependency is updated.

Reviewer: ```suggestion
The Dependency Image needs to be built only when some dependency is updated.
```
You refer to "Dependency Image" further down in the Quick Start. It's a good phrase, and introducing it here can help connect this explanation to the steps in the Quick Start.

Prestissimo dependencies change infrequently.

### Runtime Dockerfile

> This dockerfile builds Prestissimo on top of the dependency image.

Reviewer: ```suggestion
This Dockerfile builds Prestissimo on top of the dependency image.
```

> There are 2 stages. The first stage builds the Prestissimo binary.

Reviewer: ```suggestion
There are 2 stages: 
* The first stage builds the Prestissimo binary.
```
As a formatting suggestion for easier readability, make an unordered list out of these two stage descriptions?

> The second stage creates a Prestissimo runtime image from

Reviewer: ```suggestion
* The second stage creates a Prestissimo runtime image from
```

a base image and copies the binary and shared libaries that are required
for execution.

Run the following command to see the services available to build images.
```
podman compose config --services
```

## Quick Start

> ### 1. Clone this repository

Reviewer: ```suggestion
### 1. Clone the Presto repository
```
Nit: Suggest being specific when possible, especially if it's easy to do so.

```bash
git clone https://github.com/prestodb/presto
```

### 2. Build Dependency Image using docker/podman compose

> Specify the build type using BUILD_TYPE (defaults to Release)

Reviewer: ```suggestion
Specify the build type using the ``BUILD_TYPE`` (defaults to Release)
```

environment variable.

```bash
cd presto/presto-native-execution
podman compose build centos-native-dependency
```

> ### 3. Build Runtime image

Reviewer: ```suggestion
### 3. Build Runtime Image
```
Nit for consistent capitalization in the steps

> Specify the build type using BUILD_TYPE (defaults to Release)

Reviewer: ```suggestion
Specify the build type using the ``BUILD_TYPE`` (defaults to Release)
```

# Prestissimo - Dockerfile build

> 📝 _**Note:** Please post in Presto Slack if you have any questions_

There are two dockerfiles: 1) dependency.dockerfile 2) prestissimo.dockerfile
### dependency.dockefile
This dockerfile installs all the dependencies including those needed for testing.
By default, the dependencies are built in Release mode.

> This images needs to be built only when some dependency is updated.

Reviewer 2: typo: "This images needs" -> "... image ..."

> Prestissimo dependencies change very infrequently.

Reviewer 2: nit: drop "very"
